pkg/media/whip/appsrc.go

```diff
@@ -109,6 +109,8 @@ func (w *whipAppSource) Close() <-chan error {
 	logger.Debugw("WHIP app source relay Close called", "resourceID", w.resourceId, "kind", w.trackKind)
 	w.fuse.Break()
 
+	w.appSrc.EndStream()
```

Comment:
Should eos be set true here as well, or does whip not use it?

Reply:
Done

I also made a change to the WHIP handler to avoid blocking the teardown on a network read timeout.

Reply:
The whip appsrc uses a core.Fuse for that a similar purpose as the "eos" atomic in the rtmp hander, which predates core.Fuse.